fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1881,8 +1885,8 @@ public ScoredPlan plan() {
             for (KeyExpression child : indexChildren) {
                 if (!doneComparing) {
                     planChild(child);
-                    if (!comparisons.isEquality() || !foundComparison) {
-                        // Didn't add another equality; done matching filters to index.
+                    if (!comparisons.isEquality() || !foundCompleteComparison) {
+                        // Didn't add another equality or only did part of child; done matching filters to index.
```

Comment:
So if you don't match all filters, don't you have to reapply all filters that is all filters matched and unmatched? E.g. if you have a repeated `a` and you have an index on `a(FanOut).b` and a query `a.oneOfThem(b = 5 and c = 6)` wouldn't you need to do `FILTER(INDEXSCAN([5, 5], a.oneOfThem(b = 5 and c = 6)` since you will have to scope the inner `and` to one `a`?

Reply:
Yes.

In this context, `child` is a `KeyExpression`. This is the case of not matching all of the nested _keys_ to comparisons. In which case you cannot continue up a level with more unnested keys and comparisons because there is an unmatched key in the way.

I can add a test case for the query you describe, since I don't immediately see one.

Reply:
Added.